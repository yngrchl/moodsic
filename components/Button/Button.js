import React from 'react';
import PropTypes from 'prop-types';
import classNames from "classnames/bind";
import styles from './Button.module.scss';
import Link from 'next/link';

/**
 * Generic button
 * @param {string} href (optional) | redirect URL on button click
 * @param {string} text (optional) | text displayed within button
 * @param {string} type | determined styling of button. can be either "login" (green Spotify) or "error" (red error)
 */
const Button = ({ href, text, type }) => {
  const cx = classNames.bind(styles);

  if (type === "submit") {
    return (
      <button className={cx("btn", type)} type="submit">
        <h4>{text}</h4>
      </button>
    );
  }

  return(
    <Link href={href}>
      <a className={cx('btn', type)} type="submit">
        <h4>{text}</h4>
      </a>
    </Link>
)};

Button.defaultProps = {
  href: '#',
  text: ''
};

Button.propTypes = {
  href: PropTypes.string,
  text: PropTypes.string,
  type: PropTypes.oneOf(["login", "error", "submit"]).isRequired,
};

export default Button;
