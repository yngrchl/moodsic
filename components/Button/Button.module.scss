.btn {
    border-radius: 4rem;
    padding: 1rem 2rem;
    border: 1px solid white;
    display: flex;
    align-items: center;
    max-height: 4rem;

    @include at-sm {
      flex-direction: column;
    }

    &.login {
      background-color: #1DB954;
    }

    &.error {
      background-color: #ea481eff;
    }

    &.submit {
      background-color: #0070f3;
    }
}

.btn h4 {
  margin: 0;
}

.btn {
  &.login, &.error, &.submit {
    h4 {
      color: white;
    }
  }
}

.btn:hover,
.btn:focus,
.btn:active {
  &.login {
    h4 {
      color: #0d5a28;
    }

    border-color: #0d5a28;
  }

  &.error {
    h4 {
      color: #a01d1d;
    }

    border-color: #a01d1d;
  }

  &.submit {
    h4 {
      color: darkblue;
    }

    border-color: darkblue;
  }
}