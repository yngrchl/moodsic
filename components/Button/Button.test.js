import React from 'react';
import { render, screen } from '@testing-library/react';
import Button from './Button';

describe('Button', () => {
  it("renders", () => {

  });
});
