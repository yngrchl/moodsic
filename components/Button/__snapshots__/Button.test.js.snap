// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button unchanged 1`] = `
<div>
  <a
    class="btn login"
    href="#"
  >
    <h4 />
  </a>
</div>
`;
